src/parser/pseudo_instruction_parsing.rs

```diff
@@ -1237,6 +1237,973 @@ pub fn expand_pseudo_instructions_and_assign_instruction_numbers(
     }
 }
 
+///Iterates through the instruction list and translates pseudo-instructions into real instructions.
+/// LW and SW with labelled memory are not completely translated in this step because they require
+/// the address of the labelled memory to be known which is not found until after all other pseudo-instructions
+/// have been translated. Updated pseudo-instructions are added to updated_monaco_string to appear in the editor after assembly.
+/// Also ensures a syscall is at the end of the program
+pub fn expand_pseudo_instructions_and_assign_instruction_numbers_riscv(
+    instructions: &mut Vec<Instruction>,
+    data: &Vec<Data>,
+    monaco_line_info: &mut [MonacoLineInfo],
+) {
+    //figure out list of labels to be used for lw and sw labels
+    let mut list_of_labels: Vec<String> = Vec::new();
+    for instruction in instructions.clone() {
+        for label in instruction.labels {
+            list_of_labels.push(label.token.token_name);
+        }
+    }
+    for data in data {
+        list_of_labels.push(data.label.token_name.clone());
+    }
+
+    //vec_of_added_instructions is needed because of rust ownership rules. It will not let us
+    //insert into instruction_list while instruction_list is being iterated over.
+    let vec_of_added_instructions: Vec<Instruction> = Vec::new();
+
+    //iterate through every instruction and check if the operator is a pseudo-instruction
+    for (i, mut instruction) in &mut instructions.iter_mut().enumerate() {
```

Comment:
I added a function for checking operands instead of manually checking for each pseudo-instruction. The rest of the code is pseudo-instruction dependent.

Reply:
Sounds good.